api-put-object-common.go

```diff
@@ -77,10 +77,15 @@ func optimalPartInfo(objectSize int64) (totalPartsCount int, partSize int64, las
 		err = ErrEntityTooLarge(objectSize, maxMultipartPutObjectSize, "", "")
 		return
 	}
+	if int64(configuredPartSize) > objectSize {
+		err = ErrEntityTooLarge(int64(configuredPartSize), objectSize, "", "")
+		return
+	}
+
 	// Use floats for part size for all calculations to avoid
 	// overflows during float64 to int64 conversions.
 	partSizeFlt := math.Ceil(float64(objectSize / maxPartsCount))
-	partSizeFlt = math.Ceil(partSizeFlt/minPartSize) * minPartSize
+	partSizeFlt = math.Ceil(partSizeFlt/float64(configuredPartSize)) * float64(configuredPartSize)
```

Comment:
Will a user configured partsize be overridden if the object cannot be fit in 10000 parts. In that case should we fail instead of changing the partsize?

Reply:
We should fail though, but I am thinking more like it's a hint rather than an exact part size. It's like upto this partSize but we will still use the best possible for given number of parts.

Reply:
In `minio-py`, we fail if that is the case.

Reply:
Let me think more..